opentelemetry/benches/metrics.rs

```diff
@@ -28,10 +28,10 @@ fn counter_add(c: &mut Criterion) {
             counter.add(
                 1,
                 &[
-                    KeyValue::new("attribute1", "value1"),
-                    KeyValue::new("attribute2", "value2"),
-                    KeyValue::new("attribute3", "value3"),
-                    KeyValue::new("attribute4", "value4"),
+                    black_box(KeyValue::new("attribute1", "value1")),
```

Comment:
curious : what is the actual need for this here? are we worried that compiler will do some optimizations here?

Reply:
Good point. Should it be good to get the benchmark result with optimization (if any) as that would be the real scenario?

Reply:
> are we worried that compiler will do some optimizations here?

Yes, that was my motivation to add `black_box`.

Reply:
i am not convinced we need to wrap in black_box.. If we were simply creating a KeyValue, then it makes sense, as compiler might think we are not using it anywhere so try to optimize in a way that does not match real scenarios.
In this case, Counter API is used, so we probably don't need black_box.

Reply:
> compiler might think we are not using it anywhere so try to optimize in a way

The compiler could make other optimizations as well.

I benchmarked this method with and without `black_box` and there is a considerable perf difference around 32%

Without black_box: 13.523 ns
With black_box: 19.802

Reply:
we want to measure the real user scenario right? Won't compiler be doing same optimization (the one wihtthout blackbox) in their code?

Reply:
Yeah, it makes sense to mimic the real user scenario.

Reply:
Updated.